beacon-chain/casper/validator.go

```diff
@@ -25,10 +25,10 @@ func ActiveValidatorIndices(validators []*pb.ValidatorRecord, dynasty uint64) []
 
 // ExitedValidatorIndices filters out exited validators based on start and end dynasty
 // and returns their indices in a list.
-func ExitedValidatorIndices(validators []*pb.ValidatorRecord, dynasty uint64) []uint32 {
-	var indices []uint32
+func ExitedValidatorIndices(validators []*pb.ValidatorRecord) []uint32 {
+	var indices = make([]uint32, 0, len(validators))
 	for i := 0; i < len(validators); i++ {
-		if validators[i].StartDynasty < dynasty && validators[i].EndDynasty <= dynasty {
+		if validators[i].Status == uint64(params.PendingExit) {
```

Comment:
Does this work? If the initial length is 0, this will throw a NPE.

Reply:
Yep, but the slice you allocated above has a length of 0 and a capacity of `len(validators)`. So when it does enter the loop, this should throw a NPE.